harvester_e2e_tests/scenarios/test_create_vm.py

```diff
@@ -219,6 +220,51 @@ def test_create_update_vm_machine_type(request, admin_session, image, keypair,
                 utils.delete_vm(request, admin_session,
                                 harvester_api_endpoints, vm_json)
 
+
+def test_create_update_vm_user_data(request, admin_session, image, single_vm,
```

Comment:
I think we need to make the test with the `public_network` marker since we should verify that the updated VM can be SSH into with the updated user credential.